src/app/widget/widget.component.ts

```diff
@@ -80,12 +95,31 @@ export class WidgetComponent implements OnInit {
     this.widgetService.resetWidget(this.weatherWidgets);
   }
 
-  nextSlide() {
-    /*ToDo*/
+  addWidget(){
+    const newWidget = new WidgetUiMode({} as WeatherWidget);
+    this.weatherWidgets.push(newWidget);
+    if (this.weatherWidgets.length > 3){
+      this.removeLast.color = 'accent';
+      this.btnLeft.nativeElement.classList.remove('display_none');
```

Comment:
you can toggle this button via ngIf

Reply:
but for adaptive it will be better like this, on my opinion